src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/objects/inventory/Item.kt

```diff
@@ -0,0 +1,346 @@
+     * Creates an Item object from an EntityItem.
+     *
+     * @param entityItem the EntityItem.
+     */
+    constructor(entityItem: EntityItem) {
+        //#if MC<=10809
+        this.item = entityItem.entityItem.item
+        this.itemStack = entityItem.entityItem
+        //#else
+        //$$ this.item = entityItem.item.item;
+        //$$ this.itemStack = entityItem.item;
+        //#endif
+    }
+
+    /**
+     * Created an Item object from an Entity.
+     * Has to be wrapping an EntityItem.
+     *
+     * @param entity the Entity
+     */
+    constructor(entity: Entity) {
+        if (entity.entity is EntityItem) {
+            //#if MC<=10809
+            this.item = entity.entity.entityItem.item
+            this.itemStack = entity.entity.entityItem
+            //#else
+            //$$ this.item = entity.entity.item.item;
+            //$$ this.itemStack = entity.entity.item;
+            //#endif
+        } else {
+            throw IllegalArgumentException("Entity is not of type EntityItem")
+        }
+    }
+    /* End of constructors */
+
+    fun getItem() = item
+    fun getItemStack() = itemStack
+
+    /**
+     * Gets the ID of the item.
+     *
+     * @return the ID
+     */
+    fun getID() = MCItem.getIdFromItem(item)
+
+    /**
+     * Sets the stack size of the item.
+     *
+     * @param stackSize the stack size
+     * @return the Item
+     */
+    fun setStackSize(stackSize: Int) = apply {
+        itemStack = ItemStack(item, stackSize)
+    }
+
+    /**
+     * Gets the stack size of the item.
+     *
+     * @return the stack size
+     */
+    fun getStackSize(): Int {
+        //#if MC<=10809
+        return this.itemStack.stackSize
+        //#else
+        //$$ return this.itemStack.count
+        //#endif
+    }
+
+    /**
+     * Gets the item's unlocalized name.<br>
+     * Example: <code>tile.wood</code>
+     *
+     * @return the item's unlocalized name
+     */
+    fun getUnlocalizedName() = item.unlocalizedName
+
+    /**
+     * Gets the item's registry name.<br>
+     * Example: <code>minecraft:planks</code>
+     *
+     * @return the item's registry name
+     */
+    fun getRegistryName(): String {
+        //#if MC<=10809
+        return this.item.registryName
+        //#else
+        //$$ return this.item.registryName.toString();
+        //#endif
+    }
+
+    /**
+     * Gets the item's stack display name.<br>
+     * Example: <code>Oak Wood Planks</code>
+     *
+     * @return the item's stack display name
+     */
+    fun getName(): String = if (getID() == 0) "air" else itemStack.displayName
+
+    /**
+     * Returns a map of the enchantment name to level for all
+     * the enchantments on an item.
+     *
+     * @return the map of enchantments
+     */
+    fun getEnchantments(): Map<String, Int> {
+        return EnchantmentHelper.getEnchantments(itemStack).mapKeys {
+            //#if MC<=10809
+            Enchantment.getEnchantmentById(
+                //#else

```

Comment:
The indentation here should match the level of the caller, not the parameters. (i.e, shift it over)